server/npg/porchdb/data_access.py

```diff
@@ -200,10 +191,20 @@ async def update_task(self, task: Task) -> Task:
         og_task = task_result.scalar_one()
         # Check that the updated state is a valid change
         if (og_task):
+            # TODO - any rollback?
             comparable_task = og_task.convert_to_model()
             if (comparable_task.task_input_id != task.task_input_id):
                 raise Exception('Cannot change task definition. Submit a new task instead')
-            og_task.state(task.status)
+            new_status = task.status
+            # Might be the same as the old one, but save and log nevertheless
+            # in case we have some heart beat status in future.
+            og_task.state(new_status)
+            session.add(og_task)
```

Comment:
I don't think og_task needs to be added to the session. It already belongs to it because it came from the ORM. I *think*